Features/Context/PlatformUI.php

```diff
@@ -291,15 +291,19 @@ function () use ($selector, $baseElement, $checkVisibility) {
      * @param int       $iteration      Iteration number, used to control number of executions
      * @return array
      */
-    protected function getElementByText($text, $selector, $textSelector = null, $baseElement = null)
+    public function getElementByText($text, $selector, $textSelector = null, $baseElement = null)
     {
         if ($baseElement == null) {
             $baseElement = $this->getSession()->getPage();
         }
         $elements = $this->findAllWithWait($selector, $baseElement);
         foreach ($elements as $element) {
             if ($textSelector != null) {
-                $elementText = $this->findWithWait($textSelector, $element)->getText();
+                try {
+                    $elementText = $this->findWithWait($textSelector, $element)->getText();
```

Comment:
findWithWait can throw Exceptions if element does not exist or is not visible - in this case IMHO next elements from the list should be processed (if no matching element is found this function returns "false" anyway).